exporter/loadbalancingexporter/config.go

```diff
@@ -37,13 +38,27 @@ type ResolverSettings struct {
 	DNS    *DNSResolver    `mapstructure:"dns"`
 }
 
+func createDefaultResolverSettings() ResolverSettings {
+	return ResolverSettings{
+		Static: nil,
+		DNS: &DNSResolver{
+			Hostname: "",
+			Port:     "",
```

Comment:
Can the setting of the default port be moved here?

Reply:
Good idea. Will do.

Reply:
While implementing this, I noticed that I can't seem to configure this without a port anyway: there's no warning or error during load, but during execution, after the backend has been discovered (the metrics indicate an expected num_backends of 1), I see
```
{"level":"warn","ts":1653367580.393693,"caller":"zapgrpc/zapgrpc.go:191","msg":"[core] [Channel #13 SubChannel #14] grpc: addrConn.createTransport failed to connect to {\n  \"Addr\": \"10.1.0.79:4317\",\n  \"ServerName\": \"10.1.0.79:4317\",\n  \"Attributes\": null,\n  \"BalancerAttributes\": null,\n  \"Type\": 0,\n  \"Metadata\": null\n}. Err: connection error: desc = \"transport: Error while dialing dial tcp 10.1.0.79:4317: operation was canceled\"","grpc_log":true}
```
If I explicitly add a `Port: "4317"` to the Dns resolver, then it all works. Very odd.

Can someone confirm that this actually works without an explicit Port?

Reply:
Looking quickly through the exporter's code I also don't see DNSResolver setting a default port despite the readme saying it should.  Might be a bug?

Reply:
The reason I was testing this and slightly concerned is because of metrics labels: I was trying to see if in the default case the endpoint label is JUST "hostname" (or IP). In all the cases I was able to test, the endpoint label was "hostname:port". I'd hate to create a backwards compatibility issue by changing the labels in the current default case by added the port to the label (this breaking some people's dashboards or alerts or whatever).
But I haven't been able to confirm this one way or another since it doesn't appear to work in my tests..

Reply:
Can you open an issue then details your findings? The readme says there should be a default port in DNSResolver but I don't see that happening. The solution that gets discussed in the issue may be to update the README.

Reply:
Issue opened: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10321

Reply:
Back to your original request: "Can the setting of the default port be moved here?"
In trying to implement this, I'm having to go deep into tests and behavior changes. All the unit tests simply test on "endpoint-1" and "endpoint-2", which (at least for the dns-resolver) is now no longer a valid value (the port will be required, since we would add it by default). Then, by adding the port to all those tests, lots of tests that assume a hashring-lookup-result fail, because we get a different endpoint-result when searching for a specific TraceId now (I didn't get too far into why).
I'd suggest that someone with more knowledge of the code pursue this thread of explicitly adding the default port value into the DNSResolver structure.

Reply:
Yes we'll save it for a different PR.